src/utils/score_res.ts

```diff
@@ -105,6 +105,9 @@ export async function fetchPackageScore(
   ecosystem: string,
   name: string,
 ): Promise<PackageScore> {
+  if (!(ecosystem in ({} as EcosystemDestination))) {
+    throw new Error(`Invalid ecosystem: "${ecosystem}".`);
+  }
```

Comment:
Please instead check the return value of the fetch when it is not a 200 for the error field

Reply:
I tried this approach and while it works fine in dev it does not work in prod. React has a security feature that omits showing the actual error message, instead of the error message you will get 
`An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included in this error instance which may provide additional details about the nature of the error.`

Reply:
Please look at the nextjs docs for handling server errors with error.js - https://nextjs.org/docs/app/api-reference/file-conventions/error#learn-more-about-error-handling

it looks like the functionality is limited to prevent leaking server data in production. 


I would instead return { status: "ecosystem_not_found" } not throw an error

then in the page server component you can do

```
if (status === "not_found") {
      notFound();
}

if (status === "ecosystem_not_found") {
      return ... custom component to tell the user 
}